### PR TITLE
Remove dom_distiller dependency from BraveAds

### PR DESCRIPTION
**Pull request by aseren**
Resolves https://github.com/brave/brave-browser/issues/28751

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: browser/brave_ads/ads_service_factory.cc [hunk 1]
```diff
@@ -47,7 +46,6 @@ AdsServiceFactory::AdsServiceFactory()
           "AdsService",
           BrowserContextDependencyManager::GetInstance()) {
   DependsOn(NotificationDisplayServiceFactory::GetInstance());
-  DependsOn(dom_distiller::DomDistillerServiceFactory::GetInstance());
```
Comment (iefremov) [inline browser/brave_ads/ads_service_factory.cc:50]: @boocmp fyi

**Comment by iefremov** [comment]
it seems there are unused
```
app/DEPS:  "+components/dom_distiller/core/dom_distiller_switches.h",
app/brave_main_delegate.cc:#include "components/dom_distiller/core/dom_distiller_switches.h"
```

File: chromium_src/chrome/app/chrome_main_delegate.cc [hunk 2]
```diff
@@ -52,12 +52,6 @@ void AdjustSyncServiceUrlForAndroid(std::string* brave_sync_service_url) {
...
 // running those browser tests, which is not what we want.
-//
-// For instance, on DCHECK-enabled builds, not enabling the DOM distiller will
-// get all browser tests crashing when dom_distiller::RunIsolatedJavaScript()
-// gets called from AdsTabHelper::RunIsolatedJavaScript() (called via the
-// WebContentsObserver machinery), since the JS World ID won't be set yet when
-// dom_distiller::RunIsolatedJavaScript() gets called.
 absl::optional<int> ChromeMainDelegate::BasicStartupComplete() {
   BraveCommandLineHelper command_line(base::CommandLine::ForCurrentProcess());
   command_line.AppendSwitch(switches::kDisableDomainReliability);
```
Comment (boocmp) [inline chromium_src/chrome/app/chrome_main_delegate.cc:57]: Should we remove ```command_line.AppendSwitch(switches::kEnableDomDistiller);``` ?

**Comment by iefremov** [comment]
also there is a dep in `brave/browser/sources.gni`
and a strange ancient patch `patches/components-dom_distiller-content-browser-distiller_page_web_contents.cc.patch`

File: chromium_src/chrome/app/chrome_main_delegate.cc (hunk 2 above)
Comment (aseren) [reply chromium_src/chrome/app/chrome_main_delegate.cc:57]: I tried, but removing this line hides the speed-reader icon in Brave.

**Review by tmancey — APPROVED** [review]
LGTM++

**Comment by aseren** [comment]
These are still needed for `command_line.AppendSwitch(switches::kEnableDomDistiller); `
```
app/DEPS:  "+components/dom_distiller/core/dom_distiller_switches.h",
app/brave_main_delegate.cc:#include "components/dom_distiller/core/dom_distiller_switches.h"
```

**Comment by aseren** [comment]
> also there is a dep in brave/browser/sources.gni
> and a strange ancient patch patches/components-dom_distiller-content-browser-distiller_page_web_contents.cc.patch

Fixed, thanks for pointing out these. The patch becomes unused after PR https://github.com/brave/brave-core/pull/4649.

File: chromium_src/chrome/app/chrome_main_delegate.cc (hunk 2 above)
Comment (boocmp) [reply chromium_src/chrome/app/chrome_main_delegate.cc:58]: that because we have ```#define ReaderIconView SpeedreaderIconView``` :/

**Review by iefremov — APPROVED** [review]

**Review by diracdeltas — APPROVED** [review]

File: browser/brave_ads/ads_tab_helper.cc [hunk 3]
```diff
@@ -27,6 +26,15 @@
 
 namespace brave_ads {
 
+namespace {
+
+constexpr char16_t kGetDocumentHTMLScript[] =
```
Comment (bridiver) [inline browser/brave_ads/ads_tab_helper.cc:31]: We're injecting this into every page? This seems like a probable perf issue, but what exactly are we using this for anyway? If the purpose of this is to get page content, this is definitely not what we should be doing

Comment (bridiver) [reply browser/brave_ads/ads_tab_helper.cc:31]: also this appears to be happening even if ads is disabled?

**Pull request merged**